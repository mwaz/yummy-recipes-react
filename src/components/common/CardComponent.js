import React from 'react';
import { Col, Button } from 'react-bootstrap';
import PropTypes from 'prop-types';

/**
 * Component responsible for displaying both recipes and categories
 */
const CardComponent = props => {
  return (
    <Col sm={4} key={props.index}>
      {' '}
      <i type="hidden"> {props.increment}</i>
      <div className="card">
        <div className="card-title">{props.card_title}</div>
        <div className="card-text">
          <Button
            bsStyle="success"
            id="edit"
            style={{ width: '70px' }}
            onClick={props.click_edit_event}
          >
            Edit
          </Button>
          <Button
            bsStyle="danger"
            id="delete"
            style={{ marginLeft: '20px', width: '70px' }}
            onClick={props.click_delete_event}
          >
            Delete
          </Button>
        </div>
        <Button bsStyle="info" id="view" href={props.url} onClick={props.click}>
          {props.btn_name}
        </Button>
      </div>
    </Col>
  );
};

CardComponent.propTypes = {
  card_title: PropTypes.string.isRequired,
  click_delete_event: PropTypes.func.isRequired,
  click_edit_event: PropTypes.func.isRequired
};

export default CardComponent;
