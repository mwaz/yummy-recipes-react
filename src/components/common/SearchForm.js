import React from 'react';

import { Form, Row, Col, Button } from 'react-bootstrap';


/**
 * Component to handle search for both categeries and recipes
 */

const SearchForm = (props) => {
  return (
    <Form onSubmit={props.search_event}>
      <Row style={{ float: 'right' }} >
        <Col sm={8} >
          <input name="search" className="form-control" placeholder={props.search_placeholder} onKeyDown={props.name_change} onKeyPress={props.name_change} onChange={props.search} />
        </Col>
        <Col sm={3}>
          <Button id="mysearch" bsStyle="info" type="submit" onClick={props.clear}> Clear Search </Button>
        </Col>
      </Row>
    </Form>
  );
};
export default SearchForm;
