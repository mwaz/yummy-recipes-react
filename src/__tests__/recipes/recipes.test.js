import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Recipes from '../../components/recipes-front/recipes';
import SearchForm from '../../components/common/search';
import CardComponent from '../../components/common/card'
import toJson, { shallowToJson } from 'enzyme-to-json';
import { MemoryRouter } from 'react-router-dom';
import RecipeModal from '../../components/common/recipe-modal';
import DeleteComponent from '../../components/common/delete';
import Paginator from '../../components/common/paginator';

describe('Recipes component', () => {
    const wrapper = shallow( <Recipes  match={{ params: { id: 1 } }}/>
    );
    const preventDefault = jest.fn();

    it('renders properly without crashing', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('Finds a search button', () => {
        expect(wrapper.find('Button').length).toBe(1)
        
    } )
    it('it renders state initially', () => {
        expect(wrapper.state().recipe_name).toEqual('');
        expect(wrapper.state().recipe_ingredients).toEqual('');
        expect(wrapper.state().recipe_methods).toEqual('');
    });

    it('it chages states ', () => {
        wrapper.setState({
        recipe_name: 'panckakes',
        recipe_ingredients: 'milk, flour',
        recipe_methods: 'boil till ready',
        })
        expect(wrapper.state().recipe_name).toEqual('panckakes');
        expect(wrapper.state().recipe_ingredients).toEqual('milk, flour');
        expect(wrapper.state().recipe_methods).toEqual('boil till ready');
    });

    it('it adds a recipe', () => {
        const component = shallow(<RecipeModal />)
        expect(component.find('#recipe_modal_button').simulate('click'))
        expect(wrapper.instance().handleAddRecipes({preventDefault}))
    });
    
    it('it edits a recipe', () => {
        const component = shallow(<RecipeModal />)
        expect(component.find('#recipe_modal_button').simulate('click'))
        expect(wrapper.instance().editRecipe({preventDefault}))
    });
 
    it('it views recipe details', () => {
        const component = shallow(<RecipeModal />)
        expect(component.find('#recipe_modal_button').simulate('click'))
        expect(wrapper.instance().getRecipes())
    });

    it('it cancels editing a recipe', () => {
        const component = shallow(<RecipeModal />)
        expect(component.find('#cancel').simulate('click'))
        expect(wrapper.instance().editRecipe({preventDefault}))
    });

    it('it deletes a recipe', () => {
        const component = shallow(<DeleteComponent />)
        expect(component.find('#delete').simulate('click'))
        expect(wrapper.instance().handleDelete({ preventDefault }))
    });

    it('it calls the search method', () => {
        const component = shallow(<SearchForm />)
        expect(component.find('#btn-search').simulate('click'))
        expect(wrapper.instance().searchRecipes({ preventDefault }))
    });

    it('it calls the pagination requests for next page', () => {
        const component = shallow(<Paginator />)
        expect(component.find('#next').simulate('click'))
        expect(wrapper.instance().getNextPage({ preventDefault }))

    });
    it('it calls the pagination requests for previous page', () => {
        const component = shallow(<Paginator />)
        expect(component.find('#previous').simulate('click'))
        expect(wrapper.instance().getPrevPage({ preventDefault }))
    });

    it('it checks if the component has recipes', () => {
        expect(wrapper.instance().checkRecipes())
    });

    it('it shows and hides category modal', () => {
        expect(wrapper.instance().handleShow({ preventDefault }))
        expect(wrapper.instance().handleHide({ preventDefault }))
    });
    it('it renders recipes fetched', () => {
        expect(wrapper.instance().getRecipes())
    });
});