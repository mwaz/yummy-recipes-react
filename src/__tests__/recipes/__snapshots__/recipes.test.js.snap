// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes component renders properly without crashing 1`] = `
<div>
  <div
    style={
      Object {
        "background": "grey",
        "backgroundColor": "#EEEEEE",
        "color": "white",
      }
    }
  >
    <NavigationBar />
  </div>
  <div
    className="empty-div"
  />
  <div
    className="categories-parent-background"
  >
    <div
      className="categories-container"
    >
      <div
        className="grid"
      >
        <ToastContainer
          autoClose={5000}
          bodyClassName=""
          className={null}
          closeButton={
            <DefaultCloseButton
              ariaLabel="close"
            />
          }
          closeOnClick={true}
          hideProgressBar={false}
          newestOnTop={false}
          pauseOnHover={true}
          position="top-right"
          progressClassName=""
          style={null}
          toastClassName=""
          transition={[Function]}
        />
        <BreadCrumbComponent
          active_item="/Recipes/"
          href="/categories"
          item_one="Categories"
        />
        <SearchForm
          name_change={[Function]}
          search={[Function]}
          search_event={[Function]}
          search_placeholder="Search Recipes"
        />
        <Row
          bsClass="row"
          componentClass="div"
        >
          <Col
            bsClass="col"
            componentClass="div"
            sm={4}
          >
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="info"
              disabled={false}
              onClick={[Function]}
            >
               
              <span>
                 Add recipes 
              </span>
               
            </Button>
          </Col>
        </Row>
        <div>
          <RecipeModal
            actionButton={
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="success"
                disabled={false}
                id="recipe_modal_add"
                onClick={[Function]}
                type="submit"
              >
                 Add Recipe 
              </Button>
            }
            cancel_click_state={[Function]}
            hide_state={[Function]}
            ingredients_change={[Function]}
            ingredients_id="ingredients_name"
            method_state={false}
            methods_id="methods_name"
            modal_title="Add Recipe"
            name_change={[Function]}
            name_id="recipe_name"
            recipe_ingredients="Recipe Ingredients"
            recipe_methods="Recipe Methods"
            recipe_methods_change={[Function]}
            recipe_name="Recipe Name"
          />
          <DeleteComponent
            click_state={[Function]}
            form_submit={[Function]}
            item_state=""
            modal_header={[Function]}
            modal_title="Are you sure you want to delete this recipe?"
          />
          <RecipeModal
            actionButton={
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="success"
                disabled={false}
                id="recipe_modal_edit"
                onClick={[Function]}
                type="submit"
              >
                 Update 
              </Button>
            }
            cancel_click_state={[Function]}
            ingredients_change={[Function]}
            ingredients_id="ingredients_name"
            methods_id="methods_name"
            modal_header={[Function]}
            modal_title="Update Recipe"
            name_change={[Function]}
            name_id="recipe_name"
            recipe_ingredients=""
            recipe_ingredients_value=""
            recipe_methods=""
            recipe_methods_change={[Function]}
            recipe_methods_value=""
            recipe_name=""
            recipe_name_value=""
            submit_form={[Function]}
          />
          <RecipeModal
            cancel_click_state={[Function]}
            modal_header={[Function]}
            modal_title="View Recipe Details"
            readonly="readonly=\\""
            recipe_ingredients_value=""
            recipe_methods_value=""
            recipe_name_value=""
          />
        </div>
        <Row
          bsClass="row"
          componentClass="div"
        />
        <div
          className="alert alert-danger"
        >
          No recipes found on this land, kindly add them 
        </div>
        <Paginator
          next={[Function]}
          previous={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="empty-div"
  />
  <Footer />
</div>
`;
